README.rst

```diff
@@ -13,146 +13,143 @@ smart_open -- utils for streaming large files
-  >>> for line in smart_open.smart_open('hdfs://user/hadoop/my_file.txt'):
+  >>> for line in smart_open('hdfs://user/hadoop/my_file.txt'):
   ...     print line
 
   >>> # stream from HTTP
-  >>> for line in smart_open.smart_open('http://example.com/index.html'):
+  >>> for line in smart_open('http://example.com/index.html'):
   ...     print line
 
   >>> # stream from WebHDFS
-  >>> for line in smart_open.smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
+  >>> for line in smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
   ...     print line
 
   >>> # stream content *into* S3 (write mode):
-  >>> with smart_open.smart_open('s3://mybucket/mykey.txt', 'wb') as fout:
+  >>> with smart_open('s3://mybucket/mykey.txt', 'wb') as fout:
   ...     for line in ['first line', 'second line', 'third line']:
   ...          fout.write(line + '\n')
 
   >>> # stream content *into* HDFS (write mode):
-  >>> with smart_open.smart_open('hdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
+  >>> with smart_open('hdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
   ...     for line in ['first line', 'second line', 'third line']:
   ...          fout.write(line + '\n')
 
   >>> # stream content *into* WebHDFS (write mode):
-  >>> with smart_open.smart_open('webhdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
+  >>> with smart_open('webhdfs://host:port/user/hadoop/my_file.txt', 'wb') as fout:
   ...     for line in ['first line', 'second line', 'third line']:
   ...          fout.write(line + '\n')
 
-  >>> # stream from/to local compressed files:
-  >>> for line in smart_open.smart_open('./foo.txt.gz'):
+  >>> # stream using a completely custom s3 server, like s3proxy:
+  >>> for line in smart_open('s3u://user:secret@host:port@mybucket/mykey.txt'):
   ...    print line
 
-  >>> with smart_open.smart_open('/home/radim/foo.txt.bz2', 'wb') as fout:
-  ...    fout.write("some content\n")
+  >>> # you can also use a boto.s3.key.Key instance directly:
+  >>> key = boto.connect_s3().get_bucket("my_bucket").get_key("my_key")
+  >>> with smart_open(key) as fin:
+  ...     for line in fin:
+  ...         print line
 
-Since going over all (or select) keys in an S3 bucket is a very common operation,
-there's also an extra method ``smart_open.s3_iter_bucket()`` that does this efficiently,
-**processing the bucket keys in parallel** (using multiprocessing):
 
-.. code-block:: python
+Why?
+----
 
-  >>> # get all JSON files under "mybucket/foo/"
-  >>> bucket = boto.connect_s3().get_bucket('mybucket')
-  >>> for key, content in s3_iter_bucket(bucket, prefix='foo/', accept_key=lambda key: key.endswith('.json')):
-  ...     print key, len(content)
+Working with large S3 files using Amazon's default Python library, `boto <http://docs.pythonboto.org/en/latest/>`_, is a pain. Its ``key.set_contents_from_string()`` and ``key.get_contents_as_string()`` methods only work for small files (loaded in RAM, no streaming).
```

Comment:
We're using [boto3](https://boto3.readthedocs.io/en/latest/) these days, not boto.

boto3 is a bit easier to use than boto, and the API is completely different, but the sentiment of the above statement is still true.